yt_dlp/downloader/niconico.py

```diff
@@ -5,47 +5,44 @@
 from .common import FileDownloader
 from .external import FFmpegFD
 from ..networking import Request
-from ..utils import DownloadError, str_or_none, try_get
+from ..utils import DownloadError, str_or_none, truncate_string
+from ..utils.traversal import traverse_obj
 
 
 class NiconicoLiveFD(FileDownloader):
     """ Downloads niconico live without being stopped """
 
     def real_download(self, filename, info_dict):
-        video_id = info_dict['video_id']
-        ws_url = info_dict['url']
-        ws_extractor = info_dict['ws']
-        ws_origin_host = info_dict['origin']
-        live_quality = info_dict.get('live_quality', 'high')
-        live_latency = info_dict.get('live_latency', 'high')
+        video_id = info_dict['id']
+        quality, ws_extractor, ws_url = map(
+            info_dict['downloader_options'].get, ('max_quality', 'ws', 'ws_url'))
         dl = FFmpegFD(self.ydl, self.params or {})
 
         new_info_dict = info_dict.copy()
-        new_info_dict.update({
-            'protocol': 'm3u8',
-        })
+        new_info_dict['protocol'] = 'm3u8'
 
         def communicate_ws(reconnect):
             if reconnect:
```

Comment:
should we do this to support `--load-info-json`?
```suggestion
            if reconnect or not isinstance(ws_extractor, WebSocketResponse):
```

Reply:
does it mean enabling processing of an `info_dict` from JSON file while the `ws_url`’s `audience_token` is valid?
I’m not very familiar with `--load-info-json`

Reply:
The use case we would looking to support is using `--write-info-json` immediately followed by `--load-info-json`

With `--write-info-json` the info dict is run through `YoutubeDL.sanitize_info()`, which will serialize the `ws` field as its `repr()`, so it will be a string instead of a `WebSocketResponse`.

So when `ws_extractor` is a string instead of a `WebSocketResponse`, would it be possible to treat that as a "reconnect" instead of an unexpected error?